src/components/shared/Status/StatusIcon.tsx

```diff
@@ -3,11 +3,46 @@ import {
   CircleCheck,
   CircleEllipsis,
   CircleHelp,
+  CircleMinus,
   CircleX,
   RefreshCcw,
 } from "lucide-react";
 
-const StatusIcon = ({ status }: { status?: string }) => {
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+
+const StatusIcon = ({
+  status,
+  tooltip = false,
+  label = "run",
+}: {
+  status?: string;
+  tooltip?: boolean;
+  label?: "run" | "task" | "pipeline";
+}) => {
+  if (tooltip) {
+    label = label.charAt(0).toUpperCase() + label.slice(1);
+    return (
+      <Tooltip>
+        <TooltipTrigger asChild>
+          <span>
+            <Icon status={status} />
+          </span>
+        </TooltipTrigger>
+        <TooltipContent>
+          <span>{`${label} ${status?.toLowerCase() ?? "unknown"}`}</span>
+        </TooltipContent>
+      </Tooltip>
+    );
+  }
```

Comment:
```suggestion
   if (tooltip) {
    const capitalizedLabel = label.charAt(0).toUpperCase() + label.slice(1);
    const tooltipText = `${capitalizedLabel} ${status?.toLowerCase() ?? "unknown"}`;
    return (
      <Tooltip>
        <TooltipTrigger asChild>
          <span>
            <Icon status={status} />
          </span>
        </TooltipTrigger>
        <TooltipContent>
          {tooltipText}
        </TooltipContent>
      </Tooltip>
    );
  }
```

The original code is mutating the variable being passed in. lets update it to this

Reply:
A fair comment. Updated!